modules/auxiliary/admin/smb/delete_file.rb

```diff
@@ -30,21 +31,28 @@ def initialize
         [
           'mubix' # copied from hdm upload_file module
         ],
-      'License'     => MSF_LICENSE
-    )
+      'License'     => MSF_LICENSE,
+      'SessionTypes' => %w[SMB]
+      )
 
     register_options([
       OptString.new('SMBSHARE', [true, 'The name of a share on the RHOST', 'C$'])
     ])
   end
 
   def smb_delete_files
-    vprint_status("Connecting to the server...")
-    connect()
-    smb_login()
+    if session
```

Comment:
What are your thoughts on extracting out the copy/pasta into a shared method? 🤔 

Library code:

```ruby
def use_optional_smb_session_or_yield(&block)
    if session
      print_status("Using existing session #{session.sid}")
      client = session.client
      self.simple = ::Rex::Proto::SMB::SimpleClient.new(client.dispatcher.tcp_socket, client: client)
    else
      yield
    end
end
```

Calling code in module:

```ruby
use_optional_smb_session_or_yield do
  vprint_status("Connecting to the server...")
  connect()
  smb_login()
end
```

Reply:
Let's hold off on this for now, we can see how the postgres library requirements go first before we start over-optimizing an API here :+1: